src/stream/src/executor/managed_state/aggregation/extreme.rs

```diff
@@ -297,15 +316,44 @@ where
             // To future developers: please make **SURE** you have taken `EXTREME_TYPE` into
             // account. EXTREME_MIN and EXTREME_MAX will significantly impact the
             // following logic.
-            let all_data = self.keyspace.scan(self.top_n_count, epoch).await?;
-
-            for (raw_key, mut raw_value) in all_data {
-                // let mut deserializer = value_encoding::Deserializer::new(raw_value);
-                let value = deserialize_cell(&mut raw_value, &self.data_type)
-                    .map_err(StreamExecutorError::serde_error)?;
-                let key = value.clone().map(|x| x.try_into().unwrap());
-                let pks = self.serializer.get_pk(&raw_key[..])?;
-                self.top_n.insert((key, pks), value);
+
+            let all_data_iter = if let Some(gk) = self.group_key.as_ref() {
+                state_table
+                    .iter_with_pk_prefix(
+                        gk,
+                        OrderedRowSerializer::new(vec![
+                            if EXTREME_TYPE == variants::EXTREME_MAX {
+                                OrderType::Descending
+                            } else {
+                                OrderType::Ascending
+                            };
+                            self.group_key
+                                .as_ref()
+                                .map_or(0, |key| key.size())
+                        ]),
+                        epoch,
+                    )
+                    .await?
```

Comment:
use case example here. This would not compile
```
error[E0308]: `if` and `else` have incompatible types
   --> src/stream/src/executor/managed_state/aggregation/extreme.rs:339:17
    |
321 |  /             if let Some(gk) = self.group_key.as_ref() {
322 |  |                 state_table
    |  |_________________-
323 | ||                     .iter_with_pk_prefix(
324 | ||                         gk,
325 | ||                         OrderedRowSerializer::new(vec![
...   ||
336 | ||                     )
337 | ||                     .await?
    | ||___________________________- expected because of this
338 |  |             } else {
339 |  |                 state_table.iter(epoch).await?
    |  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected opaque type, found a different opaque type
340 |  |             };
    |  |_____________- `if` and `else` have incompatible types
    |
   ::: /Users/bowen/risingwave-dev/src/storage/src/table/state_table.rs:166:59
    |
166 |        pub async fn iter(&self, epoch: u64) -> StorageResult<impl RowStream<'_>> {
    |                                                              ------------------ the found opaque type
...
200 |        ) -> StorageResult<impl RowStream<'_>> {
    |                           ------------------ the expected opaque type
    |
    = note: expected opaque type `impl risingwave_storage::table::state_table::RowStream<'_>` (opaque type at </Users/bowen/risingwave-dev/src/storage/src/table/state_table.rs:200:24>)
               found opaque type `impl risingwave_storage::table::state_table::RowStream<'_>` (opaque type at </Users/bowen/risingwave-dev/src/storage/src/table/state_table.rs:166:59>)
    = note: distinct uses of `impl Trait` result in different opaque types
```

Reply:
Try this?

```rust
pub type RowStream<'a, S> = impl Stream<Item = StorageResult<Cow<'a, Row>>>;
```

...and make all `iter_xx` return this type.
```rust
pub async fn iter_with_pk_prefix(
    &self,
    pk_prefix: &Row,
    prefix_serializer: OrderedRowSerializer,
    epoch: u64,
) -> StorageResult<RowStream<'_, S>> { ... }
```

Reply:
NTFS? Or do you want to see it in this PR... But indeed it's minor changes